angular-ngrx-scss/src/app/state/profile/profile.selectors.ts

```diff
@@ -26,25 +26,41 @@ export const selectFilterBySearch = createSelector(
   (state: ProfileState) => state.sortAndFilter?.search,
 );
 
+export const selectFilterByType = createSelector(
+  selectProfileState,
+  (state: ProfileState) => state.sortAndFilter?.type || TypeFilter.All,
+);
+
 export const hasActiveSortAndFilters = createSelector(
   selectFilterBySearch,
-  Boolean,
+  selectFilterByType,
+  (search, type) => !!search || type !== TypeFilter.All,
 );
 
 export const selectRepos = createSelector(
   selectProfileState,
   selectFilterBySearch,
-  (state: ProfileState, search?: string) => {
+  selectFilterByType,
+  (state: ProfileState, search?: string, type?: string) => {
+    let filteredRepos = state.repos;
     if (search) {
-      return state.repos?.filter((item) =>
+      filteredRepos = filteredRepos?.filter((item) =>
         item.name.toLowerCase().includes(search.toLowerCase()),
       );
     }
-    return state.repos ?? [];
+    if (type !== TypeFilter.All) {
+      if (type === TypeFilter.Archived) {
+        filteredRepos = filteredRepos?.filter((item) => Boolean(item.archived));
+      }
+      if (type === TypeFilter.Forked) {
+        filteredRepos = filteredRepos?.filter((item) => Boolean(item.fork));
+      }
+    }
+    return filteredRepos ?? [];
   },
 );
 
 export const selectReposCount = createSelector(
   selectRepos,
-  (repos?: UserReposState[]) => repos?.length ?? 0,
+  (repos?: UserReposState[]) => repos?.length,
```

Comment:
question: Would this still be checking for the length of repos? Would it not need to check the length of `selectRepos` instead? 

Does `repos` always get the full repo list, or does it get updated to be the shorter list if a filter is applied?

Reply:
`repos` is updated after the filter is applied, that's why I create this selector.